tests/x11/reboot_gnome.pm

```diff
@@ -16,11 +16,24 @@ use strict;
 use testapi;
 use utils;
 
+sub assert_and_click_until_screen_change {
+    my ($mustmatch, $wait_change, $repeat) = @_;
+    $wait_change //= 1;
+    $repeat      //= 3;
+
+    for (my $i = 0; $i < $repeat; $i++) {
+        assert_and_click 'logoutdialog-reboot-highlighted';
+        if (!check_screen('logoutdialog-reboot-highlighted', $wait_change)) {
```

Comment:
It's certainly a question of *style* but you could use

```perl
last unless check_screen($mustmatch, $wait_change);
```

But I would assume the following to work better for the overall loop content:

```
wait_screen_change { assert_and_click 'logoutdialog-reboot-highlighted' };
last unless check_screen($mustmatch, 0);
```

The `wait_screen_change` should ensure that either the screen changed or we waited for some time so that also the timeout on `check_screen` would not be necessary.

Anyway, I don't think we should just accept the dialog to *not* accept a mouse press without a bug report and `record_soft_failure`